include/uk/arch/paging.h

```diff
@@ -370,14 +370,14 @@ int PAGE_Lx_IS(__pte_t pte, unsigned int lvl);
  * 64 bits for the virtual address.
  *
  * @param start the start of the virtual address range
- * @param end the end of the virtual address range
- *
+ * @param  len the length of the virtual address range, where the end address is
+ *	(start + len - 1)
  * @return a non-zero value if the addresses in the range are supported
  */
-int ukarch_vaddr_range_isvalid(__vaddr_t start, __vaddr_t end);
+int ukarch_vaddr_range_isvalid(__vaddr_t start, __sz len);
 
 #define ukarch_vaddr_isvalid(vaddr)				\
-	ukarch_vaddr_range_isvalid(vaddr, vaddr)
+	ukarch_vaddr_range_isvalid(vaddr, 1)
```

Comment:
Ouch. This looks a bit strange. And if `vaddr == vaddr_max` it will incorrectly return false. I think it would be better to directly compare against the architecturally defined limits.

Reply:
`len` will be subtracted by 1, so it will check `vaddr_max` twice, so I don't think it will be a problem. As of now, arm64 and x86 use different methods for checking valid paddr/vaddr, so this is the most portable way without exporting new interfaces.

Reply:
Agreed, but I think it still looks a bit unnatural. What I meant was because of the reasons you mention above, whether it should be implemented by the architecture-specific side of `paging.h` similarly to `ukarch_vaddr_range_isvalid()` itself, instead of being a wrapper. @marcrittinghaus, @mogasergiu any thoughts?

Reply:
>  it should be implemented by the architecture-specific side of paging.h similarly to ukarch_vaddr_range_isvalid()

I also like this better, e.g. for ARM64 use `ARM64_VADDR_VALID` and for x86 use `X86_VADDR_CANONICALIZE` :thinking:

Reply:
I updated the PR to reflect this. I added a new API `ukarch_vaddr_isvalid` that is implemented on each architecture.